tests/integration/feature_repos/test_repo_configuration.py

```diff
@@ -0,0 +1,343 @@
+    create_driver_hourly_stats_feature_view,
+)
+
+
+@dataclass(frozen=True, repr=True)
+class TestRepoConfig:
+    """
+    This class should hold all possible parameters that may need to be varied by individual tests.
+    """
+
+    provider: str = "local"
+    online_store: Union[str, Dict] = "sqlite"
+
+    offline_store_creator: str = "tests.integration.feature_repos.universal.data_sources.file.FileDataSourceCreator"
+
+    full_feature_names: bool = True
+    infer_event_timestamp_col: bool = True
+
+
+def ds_creator_path(cls: str):
+    return f"tests.integration.feature_repos.universal.data_sources.{cls}"
+
+
+DYNAMO_CONFIG = {"type": "dynamodb", "region": "us-west-2"}
+REDIS_CONFIG = {"type": "redis", "connection_string": "localhost:6379,db=0"}
+FULL_REPO_CONFIGS: List[TestRepoConfig] = [
+    # Local configurations
+    TestRepoConfig(),
+    TestRepoConfig(online_store=REDIS_CONFIG),
+    # GCP configurations
+    TestRepoConfig(
+        provider="gcp",
+        offline_store_creator=ds_creator_path("bigquery.BigQueryDataSourceCreator"),
+        online_store="datastore",
+    ),
+    TestRepoConfig(
+        provider="gcp",
+        offline_store_creator=ds_creator_path("bigquery.BigQueryDataSourceCreator"),
+        online_store=REDIS_CONFIG,
+    ),
+    # AWS configurations
+    TestRepoConfig(
+        provider="aws",
+        offline_store_creator=ds_creator_path("redshift.RedshiftDataSourceCreator"),
+        online_store=DYNAMO_CONFIG,
+    ),
+    TestRepoConfig(
+        provider="aws",
+        offline_store_creator=ds_creator_path("redshift.RedshiftDataSourceCreator"),
+        online_store=REDIS_CONFIG,
+    ),
+]
+
+
+OFFLINE_STORES: List[str] = []
+ONLINE_STORES: List[str] = []
+PROVIDERS: List[str] = []
+
+
+@dataclass
+class Environment:
+    name: str
+    test_repo_config: TestRepoConfig
+    feature_store: FeatureStore
+    data_source: DataSource
+    data_source_creator: DataSourceCreator
+
+    end_date = datetime.now().replace(microsecond=0, second=0, minute=0)
+    start_date = end_date - timedelta(days=7)
+    before_start_date = end_date - timedelta(days=365)
+    after_end_date = end_date + timedelta(days=365)
+
+    customer_entities = list(range(1001, 1110))
+    customer_df = driver_test_data.create_customer_daily_profile_df(
+        customer_entities, start_date, end_date
+    )
+    _customer_feature_view: Optional[FeatureView] = None
+
+    driver_entities = list(range(5001, 5110))
+    driver_df = driver_test_data.create_driver_hourly_stats_df(
+        driver_entities, start_date, end_date
+    )
+    _driver_stats_feature_view: Optional[FeatureView] = None
+
+    orders_df = driver_test_data.create_orders_df(
+        customers=customer_entities,
+        drivers=driver_entities,
+        start_date=before_start_date,
+        end_date=after_end_date,
+        order_count=1000,
+    )
+    _orders_table: Optional[str] = None
+
+    def customer_feature_view(self) -> FeatureView:
```

Comment:
It still seems like this universal repo related logic should be in standalone functions with one function that sets up the `Environment`, instead of being nested here.